### ISSUE TITLE
Inconsistency of timed_event in global scripts?

### ISSUE THREAD

**Issue by NovaRain** [phobos2077/sfall#277]
Here's the test script:
```pascal
#include ".\headers\define.h"
#include ".\headers\command.h"
#include ".\headers\sfall\sfall.h"

procedure start;
procedure timed_event_p_proc;

procedure start begin
   if game_loaded then begin
      add_global_timer_event(2, 65);
      display_msg("Start global timer event.");
   end
end

procedure timed_event_p_proc begin
   if (fixed_param == 65) then begin
      display_msg("Global timer event test.");
      add_global_timer_event(2, 65);
   end
end
```
If I start a new game, the script spams in the message window rather quickly (more than 270 times per minute, since the interval is only 2 ticks). But if I load a saved game and stay on the local map, the script prints the message in a much slower rate (in a lowest case it's less than 30 times per minute), sometimes it prints two messages in a row then stops a while until the next run. If I travel on the world map it runs in a normal rate (about 130 times during travel across a square).

If I save the newly started game, after re-loading it the script still runs in the normal rate. So it seems the global timer doesn't run as it should when loading existing saved games.

EDIT: I tested more of my old saved games. Looks like the results are different from save to save, in some saves the script runs in a faster/normal rate, and some are much slower. I'm guessing it might be related to critters/maps, as the faster/normal ones are saved in towns, slower ones are when I only have my party members on empty random encounter maps, the slowest one is when I just leave Arroyo and save in an empty desert map.

**Comment by FakelsHub** [comment]
Fast/slow does not make any difference, the main thing that works in sync with normal scripts and their tiks coincide.
```
G_timed_event_p_proc: 302796
timed_event_p_proc: 302796
G_timed_event_p_proc: 302798
timed_event_p_proc: 302798
G_timed_event_p_proc: 302800
timed_event_p_proc: 302800
G_timed_event_p_proc: 302802
timed_event_p_proc: 302802
G_timed_event_p_proc: 302804
timed_event_p_proc: 302804
...
G_timed_event_p_proc: 302824
timed_event_p_proc: 302824
G_timed_event_p_proc: 302826
timed_event_p_proc: 302826
G_timed_event_p_proc: 302828
timed_event_p_proc: 302828
G_timed_event_p_proc: 302830
timed_event_p_proc: 302830
```

**Comment by NovaRain** [comment]
Does that mean if there's no timer event in normal scripts currently running, it's normal that global timer by itself runs a lot slower?
I tried adding a blank timer event in obj_dude:
```pascal
   if (fixed_param == 550) then begin
      rm_fixed_timer_event(self_obj, 550);
      add_timer_event(self_obj, 20, 550); // should be 2 sec.
   end
```
And now the global timer runs more frequently when I'm alone and in the empty desert map (000).

**Comment by FakelsHub** [comment]
> Does that mean the global timer kinda depends on the execution of timer event in normal scripts?

The timed_event_p_proc procedure in the global script is run by the engine, so it depends the engine.

**Comment by NovaRain** [comment]
> 
> 
> > Does that mean the global timer kinda depends on the execution of timer event in normal scripts?
> 
> The timed_event_p_proc procedure in the global script is run by the engine, so it depends the engine.

OK. The reason why I asked is the same `timed_event_p_proc` setup:
```pascal
procedure timed_event_p_proc begin
   if (fixed_param == 65) then begin
      display_msg("Timed event triggered.");
      //add_global_timer_event(game_ticks(1), 65); // global script
      add_timer_event(self_obj, game_ticks(1), 65); // obj_dude
   end
end
```
When in a normal script (obj_dude.ssl) it runs pretty much every second no matter where I am, in towns or alone in the empty desert map. But in global scripts the setup runs much slower, especially when I'm alone in the empty desert map. Just curious about why's the difference.

**Comment by FakelsHub** [comment]
I did a test on a empty map. there is some bug.
```
=========== start
DUDE map_update_p_proc

[TimedEventScripts] run event: 933032
G_timed_event_p_proc: 933190
[TimedEventScripts] delete event: 933032

[TimedEventScripts] run event: 933192
G_timed_event_p_proc: 933370
[TimedEventScripts] delete event: 933192

DUDE map_update_p_proc

[TimedEventScripts] run event: 933372
G_timed_event_p_proc: 933530
[TimedEventScripts] delete event: 933372

=========== save
DUDE map_update_p_proc

[TimedEventScripts] run event: 933532
G_timed_event_p_proc: 933630
[TimedEventScripts] delete event: 933532

DUDE map_update_p_proc

[TimedEventScripts] run event: 933632
G_timed_event_p_proc: 933720
[TimedEventScripts] delete event: 933632

=========== load save
[TimedEventScripts] run event: 933616
G_timed_event_p_proc: 933630
[TimedEventScripts] delete event: 933616

DUDE map_update_p_proc
DUDE map_update_p_proc

[TimedEventScripts] run event: 933632
G_timed_event_p_proc: 933720
[TimedEventScripts] delete event: 933632

[TimedEventScripts] run event: 933722
G_timed_event_p_proc: 933870
[TimedEventScripts] delete event: 933722
```

**Comment by FakelsHub** [comment]
Fixed the behavior.

**Comment by NovaRain** [comment]
Fixed in commit 1ac911a.
